lib/theme_check/checks/liquid_tag.rb

```diff
@@ -24,22 +29,46 @@ def on_tag(node)
     def on_string(node)
       # Only reset the counter on outputted strings, and ignore empty line-breaks
       if node.parent.block? && !node.value.strip.empty?
-        reset_consecutive_statements
+        reset_values
       end
     end
 
     def after_document(_node)
-      reset_consecutive_statements
+      reset_values
+      @first_nodes.each do |node|
+        add_offense("Use {% liquid ... %} to write multiple tags", node: node) do |corrector|
+          consecutive = node.source[@consecutive_nodes[node.line_number][0].outer_markup_start_index, @consecutive_nodes[node.line_number][-1].outer_markup_end_index]
+          next_tag = /(?<tag_open>({%-|{%))(?<contents>(.|\n)*?)(?<tag_close>(%}|-%}))/m.match(node.source, @consecutive_nodes[node.line_number][-1].outer_markup_end_index)
+          unless next_tag.nil?
+            consecutive += "\n  #{next_tag[:contents].strip}" if next_tag[:contents].strip.start_with?("end")
+          end
+
+          consecutive.gsub!(/\n/, "")
+          consecutive.gsub!(/(\s?|\n)+(?=(-%}|%}))(-%}|%})/, "\n")
+          consecutive.gsub!(/({%-|{%)(\s?|\n)+(?=\w)/, "  ")
```

Comment:
Similarly, could we make these regex's easier to read with some names (or even composition)?